File: backend/app/schemas/campaign.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class CampaignVariantInput(BaseModel):
    variant_id: uuid.UUID
    price: Decimal = Field(gt=0)


class CampaignCreateRequest(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    product_id: uuid.UUID
    description: str | None = None
    retail_price: Decimal = Field(gt=0)
    variant_prices: list[CampaignVariantInput] = Field(min_length=1)
    starts_at: datetime | None = None
    ends_at: datetime | None = None


class CampaignUpdateRequest(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = None
    retail_price: Decimal | None = Field(default=None, gt=0)
    variant_prices: list[CampaignVariantInput] | None = None
    starts_at: datetime | None = None
    ends_at: datetime | None = None


class CampaignVariantResponse(BaseModel):
    variant_id: uuid.UUID
    variant_name: str
    price: Decimal

    model_config = {"from_attributes": True}


class CampaignResponse(BaseModel):
    id: uuid.UUID
    store_id: uuid.UUID
    product_id: uuid.UUID
    product_name: str | None = None
    title: str
    slug: str
    description: str | None = None
    design_image_url: str | None = None
    retail_price: Decimal
    status: str
    starts_at: datetime | None = None
    ends_at: datetime | None = None
    units_sold: int
    variants: list[CampaignVariantResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class CampaignListResponse(BaseModel):
    campaigns: list[CampaignResponse]
    total: int


class PublicCampaignResponse(BaseModel):
    id: uuid.UUID
    title: str
    slug: str
    description: str | None = None
    design_image_url: str | None = None
    retail_price: Decimal
    status: str
    starts_at: datetime | None = None
    ends_at: datetime | None = None
    product_name: str
    product_image_url: str | None = None
    variants: list[CampaignVariantResponse]
    store_name: str
    store_slug: str
    store_logo_url: str | None = None
    tips_enabled: bool
    tip_options: list[int | float]
    facebook_pixel_id: str | None = None
    google_analytics_id: str | None = None

    model_config = {"from_attributes": True}

File: frontend/src/lib/types/campaign.ts
export interface ProductVariant {
  id: string;
  name: string;
  sku: string | null;
  base_price: number;
}

export interface Product {
  id: string;
  name: string;
  description: string | null;
  category: string;
  image_url: string | null;
  variants: ProductVariant[];
}

export interface CampaignVariant {
  variant_id: string;
  variant_name: string;
  price: number;
}

export interface Campaign {
  id: string;
  store_id: string;
  product_id: string;
  product_name: string | null;
  title: string;
  slug: string;
  description: string | null;
  design_image_url: string | null;
  design_back_url: string | null;
  print_location: "front" | "back" | "both";
  retail_price: number;
  status: "draft" | "live" | "ended";
  starts_at: string | null;
  ends_at: string | null;
  units_sold: number;
  variants: CampaignVariant[];
  created_at: string;
  updated_at: string;
}

export interface PublicCampaign {
  id: string;
  title: string;
  slug: string;
  description: string | null;
  design_image_url: string | null;
  retail_price: number;
  status: string;
  product_name: string;
  product_image_url: string | null;
  variants: CampaignVariant[];
  store_name: string;
  store_slug: string;
  store_logo_url: string | null;
  tips_enabled: boolean;
  tip_options: number[];
  facebook_pixel_id?: string | null;
  google_analytics_id?: string | null;
}

export interface CampaignCreatePayload {
  title: string;
  product_id: string;
  description?: string;
  retail_price: number;
  print_location?: "front" | "back" | "both";
  variant_prices: { variant_id: string; price: number }[];
  starts_at?: string;
  ends_at?: string;
}

export interface CampaignUpdatePayload {
  title?: string;
  description?: string;
  retail_price?: number;
  print_location?: "front" | "back" | "both";
  variant_prices?: { variant_id: string; price: number }[];
  starts_at?: string;
  ends_at?: string;
}

export interface LiveCampaignSummary {
  id: string;
  title: string;
  slug: string;
  description: string | null;
  design_image_url: string | null;
  retail_price: number;
  product_name: string;
}
